src/core/atom.ts

```diff
@@ -25,8 +25,8 @@ export type Scope = symbol | string | number
 
 // Are there better typings?
 export type SetAtom<Update> = undefined extends Update
-  ? (update?: Update) => void | Promise<void>
-  : (update: Update) => void | Promise<void>
+  ? (update?: Update) => void
```

Comment:
Arent we losing async for write then? Maybe I'm missing something.

Reply:
The behavior doesn't change. This is more like `dispatch`ing `update`.